transmogrifier/helpers.py

```diff
@@ -83,6 +85,90 @@ def parse_xml_records(
             element.clear()
 
 
+def parse_date_from_string(
+    date_string: str,
+) -> Optional[datetime]:
+    """
+    Transform a date string into a datetime object according to one of the configured
+    OpenSearch date formats.
+
+    Args:
+        date_string: A date string.
+        source_record_id: The ID of the record being transformed.
+    """
+    for date_format in DATE_FORMATS:
+        try:
+            return datetime.strptime(date_string, date_format)
+        except ValueError:
+            pass
+    return None
+
+
+def validate_date(
+    date_string: str,
+    source_record_id: str,
+) -> bool:
+    """
+    Validate that a date string can be parsed according to one of the configured
+    OpenSearch date formats. Returns True if date string is valid or returns
+    False and logs an error.
+
+    Args:
+        date_string: A date string.
+        source_record_id: The ID of the record being transformed.
+    """
+    date_string = date_string.strip()
+    try:
```

Comment:
I'm not quite sure why this is in a try/catch block. `parse_date_from_string()` shouldn't raise a `ValueError`, it will either return a date or `None` right?